diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/generation/Generation.kt

```diff
@@ -47,6 +50,122 @@ private fun generateWarningNames() {
+                }
+                continue
+            }
+            if (iterator.hasNext() && line.trim().startsWith("```")) {
+                writer.writeCode("""\begin{lstlisting}[language=Kotlin]""")
+                line = iterator.next()
+                while (!line.trim().startsWith("```")) {
+                    writer.writeCode(line)
+                    line = iterator.next()
+                }
+                writer.writeCode("""\end{lstlisting}""")
+                continue
+            }
+
+            if (line.trim().startsWith("|")) {
+                val columnNumber = line.filter { it =='|' }.count() - 1
+                val columnWidth: Float = (15f / columnNumber) // For now it makes all column width equal
+                val createTable = "|p{${columnWidth.format(1)}cm}".repeat(columnNumber).plus("|")
+                writer.writeln("""\begin{center}""")
+                writer.writeln("""\begin{tabular}{ $createTable }""")
+                writer.writeln("""\hline""")
+                val columnNames = Regex("""[A-Za-z ]*""")
+                        .findAll(line)
+                        .filter { it.value.isNotEmpty() }
+                        .map { it.value.trim() }
+                        .toList()
+                writer.write(columnNames.joinToString(separator = "&"))
+                writer.writeln("""\\""")
+                writer.writeln("\\hline")
+                iterator.next()
+                line = iterator.next()
+                while(iterator.hasNext() && line.trim().startsWith("|")) {
+                    writer.writeln(line
+                            .replace('|', '&')
+                            .drop(1)
+                            .dropLast(1)
+                            .plus("""\\""")
+                            .replace("_", "\\_")
+                            .replace(Regex("""\(#(.*)\)"""), ""))
+                    line = iterator.next()
+                }
+                writer.writeln("""\hline""")
+                writer.writeln("\\end{tabular}")
+                writer.writeln("\\end{center}")
+            } else {
+                var correctedString = findBoldOrItalicText(line, Regex("""\*\*([^*]+)\*\*"""), FindType.BOLD)
+                correctedString = findBoldOrItalicText(correctedString, Regex("""\*([A-Za-z ]+)\*"""), FindType.ITALIC)
+                correctedString = findBoldOrItalicText(correctedString, Regex("""`([^`]*)`"""), FindType.BACKTICKS)
+                correctedString = correctedString.replace(Regex("""\(#(.*)\)"""), "")
+                correctedString = correctedString.replace("#", "\\#")
+                correctedString = correctedString.replace("&", "\\&")
+                correctedString = correctedString.replace("_", "\\_")
+                writer.writeln(correctedString)
+            }
+        }
+    }
+
+    appendixFile.removeAll(appendixFile.subList(appendixFile.indexOf("\\section*{guide}"), appendixFile.lastIndex + 1))
```

Comment:
Is `appendixFile` used before this line? Maybe, it can be read here and right after that transformed? This line looks a little messy.